### PR TITLE
The first pull request! :)

### PR DESCRIPTION
**Pull request by VittGam**
Sync with OpenWrt patches I sent to OpenWrt-Devel, and add an enhancement to /etc/profile I use on my routers. The colored prompt is not enabled by default, but it's there in case the user wants to enable it.

Cheers,
Vittorio

### CODE REVIEW

**Comment by blogic** [comment]
ack for the ramips part

**Comment by pepe2k** [comment]
"ramips: Don't use a VLAN for the single ethernet port of the A5-V11." breaks alphabetical order.

PS. Don't use dots in commit subject.

**Comment by blogic** [comment]
where does it break ordering. looks valid to me. am i blind ?

**Comment by pepe2k** [comment]
> where does it break ordering. looks valid to me. am i blind ?

https://github.com/VittGam/lede-staging/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L285

**Comment by VittGam** [comment]
Oh, _that_ order. So I believe I should move the whole block to the top instead?

By the way, I usually use ending dots in all commits subjects, not only for OpenWrt/LEDE... Is there any written FAQ with the style rules to follow? Some projects have got one, maybe we should have one too.

**Comment by pepe2k** [comment]
> Oh, that order. So I believe I should move the whole block to the top instead?

Yes, the whole file is ordered in two passes. Globally and inside every case block.

> By the way, I usually use ending dots in all commits subjects, not only for OpenWrt/LEDE... Is there any written FAQ with the style rules to follow? Some projects have got one, maybe we should have one too.

IMHO, we shouldn't. Just follow the convention, as an example take kernel repository.

BTW. This one "base-files: Enhancements to /etc/profile." I really don't like. Just another unnecessary bytes in image.

**Comment by VittGam** [comment]
> BTW. This one "base-files: Enhancements to /etc/profile." I really don't like. Just another unnecessary bytes in image.

I think that's quite useful instead. The terminal title is useful when you have a lot of terminals or terminal tabs open; the space between the directory name and the rest of the PS1 is useful for quick double-click copy; and the `ll` alias is *really* useful to me, once I got used to it I find myself quite frustrated on every system where this is not in place. And also I find the colored prompt very useful in a terminal to distinguish things at first glance, but I understand that not everybody might like it or might find it even as a distraction.

**Comment by VittGam** [comment]
> Edit: we SHOULD have style rules to follow, sorry. I was thinking only about commit subject as it's obvious for me that there shouldn't be dot at the end.

Yeah, I meant in general too :)

**Comment by pepe2k** [comment]
> I think that's quite useful instead. The terminal title is useful when you have a lot of terminals or terminal tabs open; the space between the directory name and the rest of the PS1 is useful for quick double-click copy; and the ll alias is really useful to me, once I got used to it I find myself quite frustrated on every system where this is not in place. And also I find the colored prompt very useful in a terminal to distinguish things at first glance, but I understand that not everybody might like it or might find it even as a distraction.

Not all terminal software understands that kind of fireworks. OpenWrt/Lede is IMHO focused on embedded systems. There are still devices with only 4 MB FLASH and it would be nice to still support them. With that kind of changes we are only making target image bigger without making it really more functional.

I will just put here someone else comments about similar change (http://patchwork.ozlabs.org/patch/618204/) which I fully agree with:

> Please don't force personal stylistic preferences on everyone.

Cheers

**Comment by VittGam** [comment]
I don't fully understand your point. Are you just against including the _commented_ colored line, or the rest of the patch too?

The `ll` alias is common to many Linux distributions so I fail to see how 30 bytes are going to create a space problem.

And the title in PS1 is only going to be enabled for `TERM=xterm`... Otherwise it would print garbage on the serial console, for example. But the serial console is `TERM=linux` for a reason...

**Comment by VittGam** [comment]
(By the way, I use my modded profile on all my 4MB-flash devices... it's just less than 200 bytes of an addition, and it's compressed in the squashfs so it will just not make the images grow since they usually have a blocksize of many kB... )

**Comment by pepe2k** [comment]
> I don't fully understand your point. Are you just against including the commented colored line, or the rest of the patch too?

I don't like whole change.

> (By the way, I use my modded profile on all my 4MB-flash devices... it's just less than 200 bytes of an addition, and it's compressed in the squashfs so it will just not make the images grow since they usually have a blocksize of many kB... )

So why do you want push others to use your personal stylistic preferences?

If you really see that those changes are so important, make them as configuration options, disabled by default.

**Comment by VittGam** [comment]
Sorry but I just fail to see how an alias or the title in the titlebar can be considered stylistic preferences at all. They are useful tools, found in a lot of other Linux distributions, that just simplify the everyday routine...

Anyway, I proposed the change and explained my rationale. Let's see what everybody else thinks about this...

Cheers,
Vittorio

**Comment by pepe2k** [comment]
> Sorry but I just fail to see how an alias or the title in the titlebar can be considered stylistic preferences at all. They are useful tools, found in a lot of other Linux distributions, that just simplify the everyday routine...

Don't get me wrong, from my point of view they are more useful in desktop environments, but maybe I'm just too old.

> Anyway, I proposed the change and explained my rationale. Let's see what everybody else thinks about this...

Agree, hope we will get others feedback.

Cheers!

File: package/base-files/files/bin/config_generate [hunk 1]
```diff
@@ -144,54 +144,58 @@ generate_switch_vlans_ports() {
 	#
 
 	json_get_keys roles roles
-	json_select roles
+	if [ -n "$roles" ]; then
```
Comment (jow-) [inline package/base-files/files/bin/config_generate]: In principle I agree, I just think that using `json_is_a roles array` might be a tad nicer, likewise for the other condition you introduced.

**Comment by jow-** [comment]
I'm neutral on most banner channels but I do like the window title change, the lack of an updated terminal title is whats annoying me a lot with the firmware.

**Comment by pepe2k** [comment]
Maybe make proposed changes as base-files configurable options and instead of including them inside /etc/profile file directly (especially that commended part with colors), make usage of /etc/profile.d/?

**Comment by jow-** [comment]
Making those changes configurable will add even more code though, it will also incur some overhead to each interactive login....

**Comment by thess** [comment]
This comment is more of a procedural nature -- We need to adopt some of the guidelines for patch submission and PR structuring similar to what we already do for the github packages repository and even patches to openwrt-devel.

- PR topic should be prefixed with sub-system for clarity and ease of review
- Keep patches and submissions to single subject (don't just dump your branch here)
- https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#advice-on-pull-requests
- https://dev.openwrt.org/wiki/SubmittingPatches

Please close this an re-submit these as separate topics for review.

TIA

File: package/base-files/files/bin/config_generate (hunk 1 above)
Comment (VittGam) [inline package/base-files/files/bin/config_generate]: Right, I'm going to change this. But isn't `json_is_a roles object` more appropriate here? I don't think it's an array; a JSON array key should start at zero, and should be just a progressive number. Also arrays are defined by `[]` and not by `{}`.

**Comment by VittGam** [comment]
You're right, in fact it's not been a good idea to just put all the patches here... I'm going to split this in three patch series. Thanks for the links by the way, I somehow missed them...

Cheers,
Vittorio

**Pull request closed without merging**